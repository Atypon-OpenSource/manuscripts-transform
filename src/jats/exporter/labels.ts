/*!
 * Â© 2020 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {
  ManuscriptFragment,
  ManuscriptNode,
  ManuscriptNodeType,
  schema,
} from '../../schema'
import { nodeNames } from '../../transformer/node-names'

export interface Target {
  type: string
  id: string
  label: string
  caption: string
}

interface Counter {
  label: string
  index: number
}

interface Counters {
  [key: string]: Counter
}

const labelledNodeTypes: ManuscriptNodeType[] = [
  schema.nodes.figure_element,
  schema.nodes.table_element,
  schema.nodes.equation_element,
  schema.nodes.listing_element,
  schema.nodes.box_element,
  schema.nodes.embed,
]

const excludedTypes = [schema.nodes.graphical_abstract_section]

const chooseLabel = (nodeType: ManuscriptNodeType): string => {
  return nodeNames.get(nodeType) as string
}

export const buildTargets = (
  node: ManuscriptNode | ManuscriptFragment
): Map<string, Target> => {
  const counters: Counters = {}

  for (const nodeType of labelledNodeTypes) {
    counters[nodeType.name] = {
      label: chooseLabel(nodeType), // choosing label name: "Figure", "Table", etc.
      index: 0, // TODO: use manuscript.figureElementNumberingScheme
    }
  }

  const buildLabel = (type: ManuscriptNodeType) => {
    const counter = counters[type.name]
    counter.index++
    return `${counter.label} ${counter.index}`
  }

  const targets: Map<string, Target> = new Map()

  node.descendants((node, pos, parent) => {
    if (node.type.name in counters) {
      if (parent && excludedTypes.includes(parent.type)) {
        return
      }
      const label = buildLabel(node.type)

      targets.set(node.attrs.id, {
        type: node.type.name,
        id: node.attrs.id,
        label,
        caption: node.textContent?.trim(), // TODO: HTML?
      })
    }
  })
  return targets
}
