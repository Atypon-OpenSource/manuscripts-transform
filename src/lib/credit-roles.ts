/*!
 * © 2025 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const enum CreditVocabTerm {
  Conceptualization = 'Conceptualization',
  DataCuration = 'Data curation',
  FormalAnalysis = 'Formal analysis',
  FundingAcquisition = 'Funding acquisition',
  Investigation = 'Investigation',
  Methodology = 'Methodology',
  ProjectAdministration = 'ProjectAdministration',
  Resources = 'Resources',
  Software = 'Software',
  Supervision = 'Supervision',
  Validation = 'Validation',
  Visualization = 'Visualization',
  WritingOriginalDraft = 'Writing – original draft',
  WritingReviewEditing = 'Writing – review & editing',
}

export const CRediTRoleUrls = new Map([
  [
    CreditVocabTerm.Conceptualization,
    'https://credit.niso.org/contributor-roles/conceptualization/',
  ],
  [
    CreditVocabTerm.DataCuration,
    'https://credit.niso.org/contributor-roles/data-curation/',
  ],
  [
    CreditVocabTerm.FormalAnalysis,
    'https://credit.niso.org/contributor-roles/formal-analysis/',
  ],
  [
    CreditVocabTerm.FundingAcquisition,
    'https://credit.niso.org/contributor-roles/funding-acquisition/',
  ],
  [
    CreditVocabTerm.Investigation,
    'https://credit.niso.org/contributor-roles/investigation/',
  ],
  [
    CreditVocabTerm.Methodology,
    'https://credit.niso.org/contributor-roles/methodology/',
  ],
  [
    CreditVocabTerm.ProjectAdministration,
    'https://credit.niso.org/contributor-roles/project-administration/',
  ],
  [
    CreditVocabTerm.Resources,
    'https://credit.niso.org/contributor-roles/resources/',
  ],
  [
    CreditVocabTerm.Software,
    'https://credit.niso.org/contributor-roles/software/',
  ],
  [
    CreditVocabTerm.Supervision,
    'https://credit.niso.org/contributor-roles/supervision/',
  ],
  [
    CreditVocabTerm.Validation,
    'https://credit.niso.org/contributor-roles/validation/',
  ],
  [
    CreditVocabTerm.Visualization,
    'https://credit.niso.org/contributor-roles/visualization/',
  ],
  [
    CreditVocabTerm.WritingOriginalDraft,
    'https://credit.niso.org/contributor-roles/writing-original-draft/',
  ],
  [
    CreditVocabTerm.WritingReviewEditing,
    'https://credit.niso.org/contributor-roles/writing-review-editing/',
  ],
])
