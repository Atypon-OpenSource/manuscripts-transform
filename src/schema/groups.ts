/*!
 * Â© 2020 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { NodeType } from 'prosemirror-model'

export const GROUP_BLOCK = 'block'
export const GROUP_ELEMENT = 'element'
export const GROUP_EXECUTABLE = 'executable'
export const GROUP_LIST = 'list'
export const GROUP_SECTION = 'sections'

export const hasGroup = (nodeType: NodeType, groupName: string): boolean => {
  const { group } = nodeType.spec

  if (!group) {
    return false
  }

  return group.split(/\s+/).includes(groupName)
}
