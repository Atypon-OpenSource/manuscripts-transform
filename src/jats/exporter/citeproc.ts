/*!
 * © 2025 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
//@ts-nocheck
/* eslint-disable */

import Citeproc from 'citeproc'

import { BibliographyItemAttrs } from '../../schema'

const normalizeID = (id: string) => id.replace(/:/g, '_')
const labelOnly = /^<label>.+<\/label>$/

/**
 * This is a hacky way to add JATS support to citeproc-js
 */
export const initJats = () => {
  // make sure initialization happens once
  if (Citeproc.Output.Formats.jats) {
    return
  }
  // This defines a new citeproc "format". There is no real documentation
  // for this, so the html format was copied and modified.
  Citeproc.Output.Formats.jats = {
    ...Citeproc.Output.Formats.html,
    text_escape: function (text) {
      return text ?? ''
    },
    bibstart: '<ref-list>',
    bibend: '</ref-list>',
    '@font-style/italic': '<italic>%%STRING%%</italic>',
    '@font-style/oblique': '<italic>%%STRING%%</italic>',
    '@font-style/normal': false,
    '@font-variant/small-caps': '<smallcaps>%%STRING%%</smallcaps>',
    '@font-variant/normal': false,
    '@font-weight/bold': '<bold>%%STRING%%</bold>',
    '@font-weight/normal': false,
    '@font-weight/light': false,
    '@text-decoration/none': false,
    '@text-decoration/underline': '<underline>%%STRING%%</underline>',
    '@vertical-align/sup': '<sup>%%STRING%%</sup>',
    '@vertical-align/sub': '<sub>%%STRING%%</sub>',
    '@vertical-align/baseline': false,
    // this is the entry point for rendering bibliography items. The "str" arg
    // will be the formatted bibliography item.
    '@bibliography/entry': function (state, str) {
      // the id of the bibliography item
      const id = this.system_id
      // get the bibliography item from the internal citeproc registry
      const item: BibliographyItemAttrs = state.registry.registry[id].ref
      const type = getPublicationType(item)
      str = str.trim()
      if (labelOnly.test(str)) {
        // If only a <label> is present in str, use the "literal" value of the
        // item. This is usually the case for "unstructured" references, where
        // no individual metadata is tagged. Since str includes the label, it's
        // placed before the <mixed-citation> tag.
        return `<ref id="${normalizeID(
          id
        )}">${str}<mixed-citation specific-use="unstructured-citation">${
          item.literal
        }</mixed-citation></ref>`
      } else if (str.includes('mixed-citation')) {
        // For citation styles that include a label (e.g. AMA), citeproc-js will
        // invoke both @display/left-margin and @display/right-inline, meaning
        // that str will include a <mixed-citation> tag. In that case, only the
        // <ref> tag needs to be added.
        return `<ref id="${normalizeID(id)}">${str.replace(
          '%%TYPE%%',
          type
        )}</ref>`
      } else {
        // For citation styles that do not include a label (e.g. Chicago Author),
        // citeproc-js will invoke neither @display/left-margin nor
        // @display/right-inline, meaning that str will not include a
        // <mixed-citation> tag. In that case, both <ref> and <mixed-citation>
        // need to be added.
        return `<ref id="${normalizeID(
          id
        )}"><mixed-citation publication-type="${type}">${str}</mixed-citation></ref>`
      }
    },
    '@display/block': false,
    '@display/left-margin': function (state, str) {
      return `<label>${str}</label>`
    },
    '@display/right-inline': function (state, str) {
      // this function doesn't have access to the metadat, so it's not possible
      // to determine the publication type here. Instead, a placeholder is used
      // and replaced in the @bibliography/entry function.
      return `<mixed-citation publication-type="%%TYPE%%">${str}</mixed-citation>`
    },
    '@display/indent': false,
    '@URL/true': false,
    '@DOI/true': false,
  }

  // This overrides how a person name is rendered and wraps it in
  // a <string-name> tag.
  const name = Citeproc.NameOutput.prototype._renderOnePersonalName
  Citeproc.NameOutput.prototype._renderOnePersonalName = function (...args) {
    const area = this.state.tmp.area
    const blob = name.call(this, ...args)
    if (blob && area === 'bibliography') {
      blob.strings.prefix = '<string-name>'
      blob.strings.suffix = '</string-name>'
    }
    return blob
  }

  // This overrides how the given name of a person is rendered
  // and wraps it in a <given-names> tag.
  const given = Citeproc.NameOutput.prototype._givenName
  Citeproc.NameOutput.prototype._givenName = function (...args) {
    const area = this.state.tmp.area
    const info = given.call(this, ...args)
    if (info.blob && area === 'bibliography') {
      info.blob.strings.prefix = '<given-names>'
      info.blob.strings.suffix = '</given-names>'
    }
    return info
  }

  // This overrides how the family name of a person is rendered
  // and wraps it in a <surname> tag.
  const family = Citeproc.NameOutput.prototype._familyName
  Citeproc.NameOutput.prototype._familyName = function (...args) {
    const area = this.state.tmp.area
    const blob = family.call(this, ...args)
    if (blob && area === 'bibliography') {
      blob.strings.prefix = '<surname>'
      blob.strings.suffix = '</surname>'
    }
    return blob
  }
}

const namesWrapper = (type: string) => (str: string) =>
  `<person-group person-group-type="${type}">${str}</person-group>`

// Even though the CSL data model specifies a date type, citeproc-js processes
// date fields and replaces their value with an object with the structure:
// {
//    year: number
//    month: number
//    day: number
// }
// see CSL.Engine.prototype.dateParseArray for more details.
const formatDate = (date) => {
  let output = date.year
  if (date.month) {
    output += '-' + String(date.month).padStart(2, '0')
    if (date.day) {
      output += '-' + String(date.day).padStart(2, '0')
    }
  }
  return output
}

const getPublicationType = (item: BibliographyItemAttrs) => {
  switch (item.type) {
    case 'article-journal':
      return 'journal'
    case 'webpage':
      return 'page'
    case 'dataset':
      return 'data'
    default:
      return item.type
  }
}

const wrappers = {
  // this is a name field, so other processing is required. See the "initJats"
  // function for more info.
  author: namesWrapper('author'),
  // Unlike other date fields, where a formatted date string is wrapped in
  // a tag with the "iso-8601-date" attribute, the "issued" date components
  // are individually wrapped in year/month/day tags. For the time being,
  // only the year component is processed.
  issued: (str: string, item: BibliographyItemAttrs) =>
    str.replace(item.issued.year, `<year>${item.issued.year}</year>`),
  'container-title': (str: string) => `<source>${str}</source>`,
  volume: (str: string) => `<volume>${str}</volume>`,
  issue: (str: string) => `<issue>${str}</issue>`,
  supplement: (str: string) => `<supplement>${str}</supplement>`,
  page: (str: string) => {
    const parts = str.split(/[-–]/)
    if (parts.length === 2) {
      const fpage = parts[0].trim()
      const lpage = parts[1].trim()
      return `<fpage>${fpage}</fpage><lpage>${lpage}</lpage>`
    } else {
      return `<fpage>${str.trim()}</fpage>`
    }
  },
  title: (str: string, item: BibliographyItemAttrs) => {
    const type = item.type
    switch (type) {
      case 'dataset':
        return `<data-title>${str}</data-title>`
      case 'article-journal':
      case 'preprint':
        return `<article-title>${str}</article-title>`
      default:
        return `<part-title>${str}</part-title>`
    }
  },
  std: (str: string) => `<pub-id pub-id-type="std-designation">${str}</pub-id>`,
  'collection-title': (str: string) => `<series>${str}</series>`,
  edition: (str: string) => `<edition>${str}</edition>`,
  publisher: (str: string) => `<publisher-name>${str}</publisher-name>`,
  'publisher-place': (str: string) => `<publisher-loc>${str}</publisher-loc>`,
  event: (str: string) => `<conf-name>${str}</conf-name>`,
  'event-place': (str: string) => `<conf-loc>${str}</conf-loc>`,
  'number-of-pages': (str: string) => `<size units="pages">${str}</size>`,
  institution: (str: string) => `<institution>${str}</institution>`,
  locator: (str: string) => `<elocation-id>${str}</elocation-id>`,
  // this is a name field, so other processing is required. See the "initJats"
  // function for more info.
  editor: namesWrapper('editor'),
  accessed: (str: string, item: BibliographyItemAttrs) =>
    `<date-in-citation content-type="access-date" iso-8601-date="${formatDate(
      item.accessed
    )}">${str}</date-in-citation>`,
  'event-date': (str: string, item: BibliographyItemAttrs) =>
    `<conf-date iso-8601-date="${formatDate(
      item['event-date']
    )}">${str}</conf-date>`,
  DOI: (str: string) => `<pub-id pub-id-type="doi">${str}</pub-id>`,
  URL: (str: string) =>
    `<ext-link ext-link-type="uri" xlink:href="${str}">${str}</ext-link>`,
}

/**
 * A function that formats the CSL variables into JATS. This is a simple operation
 * in most cases: the passed string is wrapped with the appropriate tag
 * as is. However, some variables require special handling.
 */
export const jatsVariableWrapper: Citeproc.VariableWrapper = (
  params,
  pre,
  str,
  post
) => {
  if (str && params.context === 'bibliography') {
    const name = params.variableNames[0]
    const fn = wrappers[name]
    if (fn) {
      str = fn(str, params.itemData)
    }
  }
  return `${pre}${str ?? ''}${post}`
}
