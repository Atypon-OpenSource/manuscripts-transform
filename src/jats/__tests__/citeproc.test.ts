/*!
 * Â© 2025 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import Citeproc from 'citeproc'

import { buildCiteprocCitation } from '../../lib/citeproc'
import { BibliographyItemAttrs } from '../../schema'
import { initJats, jatsVariableWrapper } from '../exporter/citeproc'
import {ama, chicago, DEFAULT_CSL_OPTIONS} from './citations'

const bib1: BibliographyItemAttrs = {
  id: 'bib1',
  type: 'article-journal',
  author: [
    { given: 'John', family: 'Doe' },
    { given: 'Jane', family: 'Smith' },
  ],
  DOI: '10.1000/xyz123',
  issued: { 'date-parts': [[2021]] },
  'container-title': 'AI Research Journal',
  volume: '18',
  issue: '2',
  supplement: 'Suppl. 1',
  page: '45-60',
  title: 'Advancements in Artificial Intelligence',
  locator: 'e012345',
  URL: 'https://doi.org/10.1000/xyz123',
}

const bib2: BibliographyItemAttrs = {
  id: 'bib2',
  type: 'book',
  author: [
    { given: 'Michael', family: 'Brown' },
    { given: 'Anna', family: 'Green' },
  ],
  issued: { 'date-parts': [[2020]] },
  'container-title': 'Introduction to Data Science',
  volume: '1',
  'collection-title': 'Data Science Series',
  edition: '2nd',
  publisher: 'Springer',
  'publisher-place': 'New York, USA',
  editor: [
    { given: 'David', family: 'Williams' },
    { given: 'Sophia', family: 'Taylor' },
  ],
  'number-of-pages': '350',
  URL: 'https://doi.org/10.1000/xyz123',
}

const bib3: BibliographyItemAttrs = {
  id: 'bib3',
  type: 'chapter',
  author: [{ given: 'Susan', family: 'Miller' }],
  issued: { 'date-parts': [[2022]] },
  'container-title': 'Handbook of AI',
  page: '105-120',
  title: 'Machine Learning Applications',
  publisher: 'Oxford University Press',
}

const bib4: BibliographyItemAttrs = {
  id: 'bib4',
  type: 'confproc',
  author: [{ given: 'Emily', family: 'Johnson' }],
  issued: { 'date-parts': [[2019]] },
  'container-title':
    'Proceedings of the International Conference on Machine Learning',
  page: '223-230',
  title: 'Deep Learning in Image Processing',
  publisher: 'IEEE',
  'publisher-place': 'New York, USA',
  event: 'ICML 2019',
  'event-place': 'San Francisco, USA',
  'event-date': { 'date-parts': [[2019, 6, 20]] },
  URL: 'https://icml2019.org/papers/223',
}

const bib5: BibliographyItemAttrs = {
  id: 'bib5',
  type: 'thesis',
  author: [{ given: 'Daniel', family: 'Garcia' }],
  DOI: '10.1234/thesis2021',
  issued: { 'date-parts': [[2021]] },
  title: 'Neural Networks for Natural Language Processing',
  institution: 'University of California, Berkeley',
  'number-of-pages': '350',
  URL: 'https://escholarship.org/uc/item/xyz123',
  comment:
    'Located at: Modern Manuscripts Collection,\n' +
    '                        History of Medicine Division, National Library of\n' +
    '                        Medicine, Bethesda, MD; MS F 179',
}

const bib6: BibliographyItemAttrs = {
  id: 'bib6',
  type: 'webpage',
  author: [{ given: 'Sarah', family: 'Jones' }],
  issued: { 'date-parts': [[2023]] },
  'container-title': 'Data Science Blog',
  title: 'Big Data Trends in 2023',
  accessed: { 'date-parts': [[2024, 2, 15]] },
  URL: 'https://www.datascienceblog.com/big-data-2023',
}

const bib7: BibliographyItemAttrs = {
  id: 'bib7',
  type: 'other',
  author: [{ given: 'Lisa', family: 'Adams' }],
  DOI: undefined,
  issued: { 'date-parts': [[2022]] },
  'container-title': 'Internal Research Report',
  title: 'Private Research Report on AI Ethics',
  publisher: 'AI Ethics Initiative',
}

const buildCitation = (id: string) =>
  buildCiteprocCitation({
    id: `${id}-citation`,
    rids: [id],
  })

describe('Export formatted references', () => {
  initJats()

  const data = [
    ['AMA', ama],
    ['Chicago Author Date', chicago],
  ]

  test.each(data)('Export references in %s style', async (name, style) => {
    const bibitems: Map<string, BibliographyItemAttrs> = new Map()
    bibitems.set(bib1.id, bib1)
    bibitems.set(bib2.id, bib2)
    bibitems.set(bib3.id, bib3)
    bibitems.set(bib4.id, bib4)
    bibitems.set(bib5.id, bib5)
    bibitems.set(bib6.id, bib6)
    bibitems.set(bib7.id, bib7)

    const citations: Citeproc.Citation[] = [
      buildCitation(bib1.id),
      buildCitation(bib2.id),
      buildCitation(bib3.id),
      buildCitation(bib4.id),
      buildCitation(bib5.id),
      buildCitation(bib6.id),
      buildCitation(bib7.id),
    ]

    initJats()
    const engine = new Citeproc.Engine(
      {
        retrieveLocale: () => DEFAULT_CSL_OPTIONS.locale,
        retrieveItem: (id: string) => {
          const item = bibitems.get(id)
          if (!item) {
            throw Error(`Missing bibliography item with id ${id}`)
          }
          return item as CSL.Data
        },
        variableWrapper: jatsVariableWrapper,
      },
      style
    )
    engine.setOutputFormat('jats')

    const citationTexts = engine.rebuildProcessorState(citations)
    const [meta, bibliography] = engine.makeBibliography()

    for (const [id, _, text] of citationTexts) {
      expect(text).toMatchSnapshot(id)
    }

    for (let i = 0; i < bibliography.length; i++) {
      const id = meta.entry_ids[i][0]
      const fragment = bibliography[i]
      expect(fragment).toMatchSnapshot(id)
    }
  })
})
