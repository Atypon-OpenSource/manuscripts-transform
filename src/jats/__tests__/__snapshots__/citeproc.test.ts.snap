// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Export formatted references Export references in AMA style: bib1 1`] = `"<ref id=\\"bib1\\"><label>1. </label><mixed-citation publication-type=\\"journal\\"><person-group person-group-type=\\"author\\"><string-name><surname>Doe</surname> <given-names>J</given-names></string-name>, <string-name><surname>Smith</surname> <given-names>J</given-names></string-name></person-group>. <article-title>Advancements in Artificial Intelligence</article-title>. <source><italic>AI Research Journal</italic></source>. <year>2021</year>;<volume>18</volume><issue>(2)</issue>:<fpage>45</fpage><lpage>60</lpage>. doi:<pub-id pub-id-type=\\"doi\\">10.1000/xyz123</pub-id></mixed-citation></ref>"`;

exports[`Export formatted references Export references in AMA style: bib1-citation 1`] = `"<sup>1</sup>"`;

exports[`Export formatted references Export references in AMA style: bib2 1`] = `"<ref id=\\"bib2\\"><label>2. </label><mixed-citation publication-type=\\"book\\"><person-group person-group-type=\\"author\\"><string-name><surname>Brown</surname> <given-names>M</given-names></string-name>, <string-name><surname>Green</surname> <given-names>A</given-names></string-name></person-group>. Vol <volume>1</volume>. <edition>2nd</edition> ed. (<person-group person-group-type=\\"editor\\"><string-name><surname>Williams</surname> <given-names>D</given-names></string-name>, <string-name><surname>Taylor</surname> <given-names>S</given-names></string-name>, eds</person-group>.). <publisher-loc>New York, USA</publisher-loc>: <publisher-name>Springer</publisher-name>; <year>2020</year>. <ext-link ext-link-type=\\"uri\\" xlink:href=\\"https://doi.org/10.1000/xyz123\\">https://doi.org/10.1000/xyz123</ext-link>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in AMA style: bib2-citation 1`] = `"<sup>2</sup>"`;

exports[`Export formatted references Export references in AMA style: bib3 1`] = `"<ref id=\\"bib3\\"><label>3. </label><mixed-citation publication-type=\\"chapter\\"><person-group person-group-type=\\"author\\"><string-name><surname>Miller</surname> <given-names>S</given-names></string-name></person-group>. <part-title>Machine Learning Applications</part-title>. In: <source><italic>Handbook of AI</italic></source>. <publisher-name>Oxford University Press</publisher-name>; <year>2022</year>:<fpage>105</fpage><lpage>120</lpage>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in AMA style: bib3-citation 1`] = `"<sup>3</sup>"`;

exports[`Export formatted references Export references in AMA style: bib4 1`] = `"<ref id=\\"bib4\\"><label>4. </label><mixed-citation publication-type=\\"confproc\\"><person-group person-group-type=\\"author\\"><string-name><surname>Johnson</surname> <given-names>E</given-names></string-name></person-group>. <part-title>Deep Learning in Image Processing</part-title>. <source><italic>Proceedings of the International Conference on Machine Learning</italic></source>. <year>2019</year>:<fpage>223</fpage><lpage>230</lpage>. <ext-link ext-link-type=\\"uri\\" xlink:href=\\"https://icml2019.org/papers/223\\">https://icml2019.org/papers/223</ext-link>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in AMA style: bib4-citation 1`] = `"<sup>4</sup>"`;

exports[`Export formatted references Export references in AMA style: bib5 1`] = `"<ref id=\\"bib5\\"><label>5. </label><mixed-citation publication-type=\\"thesis\\"><person-group person-group-type=\\"author\\"><string-name><surname>Garcia</surname> <given-names>D</given-names></string-name></person-group>. <part-title>Neural Networks for Natural Language Processing</part-title>. <year>2021</year>. doi:<pub-id pub-id-type=\\"doi\\">10.1234/thesis2021</pub-id></mixed-citation></ref>"`;

exports[`Export formatted references Export references in AMA style: bib5-citation 1`] = `"<sup>5</sup>"`;

exports[`Export formatted references Export references in AMA style: bib6 1`] = `"<ref id=\\"bib6\\"><label>6. </label><mixed-citation publication-type=\\"page\\"><person-group person-group-type=\\"author\\"><string-name><surname>Jones</surname> <given-names>S</given-names></string-name></person-group>. <part-title>Big Data Trends in 2023</part-title>. <source>Data Science Blog</source>. <ext-link ext-link-type=\\"uri\\" xlink:href=\\"https://www.datascienceblog.com/big-data-2023\\">https://www.datascienceblog.com/big-data-2023</ext-link>. Published <year>2023</year>. Accessed <date-in-citation content-type=\\"access-date\\" iso-8601-date=\\"2024-02-15\\">February 15, 2024</date-in-citation>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in AMA style: bib6-citation 1`] = `"<sup>6</sup>"`;

exports[`Export formatted references Export references in AMA style: bib7 1`] = `"<ref id=\\"bib7\\"><label>7. </label><mixed-citation publication-type=\\"other\\"><person-group person-group-type=\\"author\\"><string-name><surname>Adams</surname> <given-names>L</given-names></string-name></person-group>. <part-title>Private Research Report on AI Ethics</part-title>. <source><italic>Internal Research Report</italic></source>. <year>2022</year>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in AMA style: bib7-citation 1`] = `"<sup>7</sup>"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib1 1`] = `"<ref id=\\"bib1\\"><mixed-citation publication-type=\\"journal\\"><person-group person-group-type=\\"author\\"><string-name><surname>Doe</surname>, <given-names>John</given-names></string-name>, and <string-name><given-names>Jane</given-names> <surname>Smith</surname></string-name></person-group>. <year>2021</year>. <article-title>“Advancements in Artificial Intelligence.”</article-title> <source><italic>AI Research Journal</italic></source> <volume>18</volume> (<issue>2</issue>): <fpage>45</fpage><lpage>60</lpage>. <pub-id pub-id-type=\\"doi\\">https://doi.org/10.1000/xyz123</pub-id>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib1-citation 1`] = `"(Doe and Smith 2021)"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib2 1`] = `"<ref id=\\"bib2\\"><mixed-citation publication-type=\\"book\\"><person-group person-group-type=\\"author\\"><string-name><surname>Brown</surname>, <given-names>Michael</given-names></string-name>, and <string-name><given-names>Anna</given-names> <surname>Green</surname></string-name></person-group>. <year>2020</year>. <person-group person-group-type=\\"editor\\">Edited by <string-name><given-names>David</given-names> <surname>Williams</surname></string-name> and <string-name><given-names>Sophia</given-names> <surname>Taylor</surname></string-name></person-group>. <source><italic>Introduction to Data Science</italic></source>. <edition>2nd</edition> ed. Vol. <volume>1</volume>. <series>Data Science Series</series>. <publisher-loc>New York, USA</publisher-loc>: <publisher-name>Springer</publisher-name>. <ext-link ext-link-type=\\"uri\\" xlink:href=\\"https://doi.org/10.1000/xyz123\\">https://doi.org/10.1000/xyz123</ext-link>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib2-citation 1`] = `"(Brown and Green 2020)"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib3 1`] = `"<ref id=\\"bib3\\"><mixed-citation publication-type=\\"chapter\\"><person-group person-group-type=\\"author\\"><string-name><surname>Miller</surname>, <given-names>Susan</given-names></string-name></person-group>. <year>2022</year>. <part-title>“Machine Learning Applications.”</part-title> In <source><italic>Handbook of AI</italic></source>, <fpage>105</fpage><lpage>20</lpage>. <publisher-name>Oxford University Press</publisher-name>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib3-citation 1`] = `"(Miller 2022)"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib4 1`] = `"<ref id=\\"bib4\\"><mixed-citation publication-type=\\"confproc\\"><person-group person-group-type=\\"author\\"><string-name><surname>Johnson</surname>, <given-names>Emily</given-names></string-name></person-group>. <year>2019</year>. <part-title>“Deep Learning in Image Processing.”</part-title> <source><italic>Proceedings of the International Conference on Machine Learning</italic></source>. <publisher-loc>New York, USA</publisher-loc>: <publisher-name>IEEE</publisher-name>. <ext-link ext-link-type=\\"uri\\" xlink:href=\\"https://icml2019.org/papers/223\\">https://icml2019.org/papers/223</ext-link>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib4-citation 1`] = `"(Johnson 2019)"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib5 1`] = `"<ref id=\\"bib5\\"><mixed-citation publication-type=\\"thesis\\"><person-group person-group-type=\\"author\\"><string-name><surname>Garcia</surname>, <given-names>Daniel</given-names></string-name></person-group>. <year>2021</year>. <part-title>“Neural Networks for Natural Language Processing.”</part-title> <pub-id pub-id-type=\\"doi\\">https://doi.org/10.1234/thesis2021</pub-id>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib5-citation 1`] = `"(Garcia 2021)"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib6 1`] = `"<ref id=\\"bib6\\"><mixed-citation publication-type=\\"page\\"><person-group person-group-type=\\"author\\"><string-name><surname>Jones</surname>, <given-names>Sarah</given-names></string-name></person-group>. <year>2023</year>. <part-title>“Big Data Trends in 2023.”</part-title> <source>Data Science Blog</source>. <year>2023</year>. <ext-link ext-link-type=\\"uri\\" xlink:href=\\"https://www.datascienceblog.com/big-data-2023\\">https://www.datascienceblog.com/big-data-2023</ext-link>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib6-citation 1`] = `"(Jones 2023)"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib7 1`] = `"<ref id=\\"bib7\\"><mixed-citation publication-type=\\"other\\"><person-group person-group-type=\\"author\\"><string-name><surname>Adams</surname>, <given-names>Lisa</given-names></string-name></person-group>. <year>2022</year>. <part-title>“Private Research Report on AI Ethics.”</part-title> <source><italic>Internal Research Report</italic></source>. <publisher-name>AI Ethics Initiative</publisher-name>.</mixed-citation></ref>"`;

exports[`Export formatted references Export references in Chicago Author Date style: bib7-citation 1`] = `"(Adams 2022)"`;
